packages/node/src/generators/application/application.ts

```diff
@@ -290,14 +292,28 @@ function addProjectDependencies(
 }
 
 function updateTsConfigOptions(tree: Tree, options: NormalizedSchema) {
-  // updatae tsconfig.app.json to typecheck default exports  https://www.typescriptlang.org/tsconfig#esModuleInterop
-  updateJson(tree, `${options.appProjectRoot}/tsconfig.app.json`, (json) => ({
-    ...json,
-    compilerOptions: {
-      ...json.compilerOptions,
-      esModuleInterop: true,
-    },
-  }));
+  if (options.framework && options?.bundler === 'esbuild') {
+    // updatae tsconfig.app.json to typecheck default exports  https://www.typescriptlang.org/tsconfig#esModuleInterop
+    updateJson(tree, `${options.appProjectRoot}/tsconfig.app.json`, (json) => ({
```

Comment:
I think this may have been a mistake in the past. We should always update the `compilerOptions` on `tsconfig.json` not `tsconfig.app.json`.

The reason is that other tsconfig files such as `tsconfig.spec.json` will also need those options.

We should update it to be:

```
if (options.rootProject) {
  // ... 
} else {
  // ...
}
```

I don't think need to check framework or bundler, unless the framework needs other entries in types, like `types: ['node', 'express']`.